packages/client-beta/src/views/MenuBar/components/TimeWidget/TimeWidget.tsx

```diff
@@ -0,0 +1,33 @@
+import { createSignal, onCleanup, onMount } from 'solid-js';
+
+const FIVE_SECONDS = 5 * 1000;
+
+const formatter = new Intl.DateTimeFormat('pl-PL', {
+  hour: '2-digit',
+  minute: '2-digit',
+});
+
+function getFormattedTime(): string {
+  return formatter.format(new Date());
+}
+
+function TimeWidget() {
+  const [time, setTime] = createSignal(getFormattedTime());
+  const [intervalId, setIntervalId] = createSignal<NodeJS.Timer>();
+
+  onMount(() => {
+    const interval = setInterval(() => {
+      setTime(getFormattedTime());
+    }, FIVE_SECONDS);
```

Comment:
Dlaczego co 5 sekund? Wydaje mi się, że po otwarciu strony powinien od razu pobrać aktualną godzinę, a potem aktualizować go interwałem co minutę, bo tylko minuty wyświetlamy

Reply:
Powinno być co sekunde tak na prawdę :D. Co minutę to zakładając, że godzinę pobierzesz o równej minucie (np. 15:00:00), bo jak pobierzesz na przykład 15:00:59 to do 15:01:59 będziesz miał nieaktualną godzinę. Te 5 sekund to rzeczywiście taka znikąd liczba, ale stwierdziłem, że to tylko ficzer wizualny i te ewentualne maks 5 sekund opóźnienia w zegarku nie będzie tak bolało, a co sekundę uruchamiać Jsa dla tego ficzerka wydawało mi się overkill. 

Podsumowując - chciałem jakoś to zoptymalizować, żeby nie trzeba było co sekundę tego robić. Czy taka optymalizacja rzeczywiście jest konieczna to nie wiem w sumie :D